src/cache/cache_las.c

```diff
@@ -726,24 +739,15 @@ __wt_las_insert_block(WT_SESSION_IMPL *session, WT_CURSOR *cursor,
 
 err:	/* Resolve the transaction. */
 	if (local_txn) {
-		if (ret == 0) {
-			/*
-			 * Adjust the entry count.
-			 *
-			 * For inserts, we increment before committing.  As
-			 * soon as we commit, sweep could catch up and remove
-			 * the block, and we don't want the count to underflow.
-			 * In the unlikely event that the commit fails, roll
-			 * back the increment.
-			 */
-			__wt_atomic_add64(
-			    &conn->cache->las_entry_count, insert_cnt);
-			if ((ret = __wt_txn_commit(las_session, NULL)) != 0)
-				__wt_cache_decr_check_uint64(session,
-				    &conn->cache->las_entry_count,
-				    insert_cnt, "lookaside entry count");
-		} else
+		if (ret == 0)
+			WT_TRET(__wt_txn_commit(las_session, NULL));
+		else
 			WT_TRET(__wt_txn_rollback(las_session, NULL));
+
+		/* Adjust the entry count. */
+		if (ret == 0)
+			__wt_atomic_add64(
+			    &conn->cache->las_insert_count, insert_cnt);
```

Comment:
`__wt_atomic_add64()` returns a value so needs a void cast (there are a couple of other instances, too).